tests/implicit/test-stratigraphic-units-stack.cpp

```diff
@@ -65,7 +67,16 @@ void test_stratigraphic_stack()
     OPENGEODE_EXCEPTION(
         units_stack.above( units_stack.above( unit0 ).value() ).value()
             == unit1,
-        "[TEST] Horizon 1 should be found from horizon 0." );
+        "[TEST] Unit 1 should be found from unit 0." );
+    const auto insertion_info = stack_builder.add_horizon_in_stratigraphic_unit(
+        units_stack.stratigraphic_unit( unit1 ) );
+    OPENGEODE_EXCEPTION( units_stack.above( hor1 ).value()
+                             == insertion_info.strati_unit_under_id,
+        "[TEST] New bottom unit should be found above horizon 1." );
+    OPENGEODE_EXCEPTION(
+        units_stack.above( units_stack.above( hor1 ).value() ).value()
+            == insertion_info.new_horizon_id,
+        "[TEST] New horizon should be found from horizon 1." );
```

Comment:
```suggestion
        "[Test] Unit 1 should be found from unit 0." );
    const auto insertion_info = stack_builder.add_horizon_in_stratigraphic_unit(
        units_stack.stratigraphic_unit( unit1 ) );
    OPENGEODE_EXCEPTION( units_stack.above( hor1 ).value()
                             == insertion_info.strati_unit_under_id,
        "[Test] New bottom unit should be found above horizon 1." );
    OPENGEODE_EXCEPTION(
        units_stack.above( units_stack.above( hor1 ).value() ).value()
            == insertion_info.new_horizon_id,
        "[Test] New horizon should be found from horizon 1." );
```
Check elsewhere in the file